app/src/main/java/com/github/epfl/meili/messages/ChatMessageViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.github.epfl.meili.messages
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.github.epfl.meili.models.ChatMessage
+import java.util.*
+
+object ChatMessageViewModel : ViewModel(),
+    Observer {
+
+    lateinit var database: MessageDatabase
+
+    private val _messages = MutableLiveData<List<ChatMessage>?>()
+    val messages: LiveData<List<ChatMessage>?> = _messages
+
+
+
+    fun setMessageDatabase(database: MessageDatabase){
+        this.database = database
+        database.addObserver(this)
+    }
+
+    fun addMessage(text: String, fromId: String, toId: String, timeStamp: Long) {
+        val message = ChatMessage(
+            text,
+            fromId,
```

Comment:
for now, the fromId is usually "" until it arrives to the MessageDatabase service, there it gets populated. In the case of the Firebase Message Database Adapter it replaces it with Firebase.auth. This is done so that the tests don't call Firebase.auth when using the mock database